Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,234 @@
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(caseDescription, forKey: .caseDescription)
+    switch self {
+    case let .simple(name):
+      try container.encode(name, forKey: .text)
+    case let .unknown(text):
+      try container.encode(text, forKey: .text)
+    case let .optional(type),
+         let .array(type),
+         let .implicitlyUnwrapped(type):
+      try container.encode(type, forKey: .typeDescription)
+    case let .tuple(types),
+         let .composition(types):
+      try container.encode(types, forKey: .typeDescriptions)
+    case let .dictionary(key, value):
+      try container.encode(key, forKey: .typeDescriptionKey)
+      try container.encode(value, forKey: .typeDescriptionValue)
+    case let .member(name, baseType):
+      try container.encode(name, forKey: .text)
+      try container.encode(baseType, forKey: .typeDescription)
+    }
+  }
+
+  enum CodingKeys: String, CodingKey {
+    case caseDescription
+    case text
+    case typeDescription
+    case typeDescriptions
+    case typeDescriptionKey
+    case typeDescriptionValue
+  }
+
+  enum CodingError: Error {
+    case unknownCase
+  }
+
+  /// - Parameters:
+  ///   - name: The simple name of the returned type.
+  ///   - parent: The parent type for the returned type.
+  /// - Returns: Returns a type description of case `.member` with the given name as the name and the receiver as the base type.
+  /// - Note: This method only makes sense when the `parent` is of case  `simple`, `member`, `optional`, and `implicitlyUnwrapped`.
+  static func createTypeWithName(_ name: String, parent: TypeDescription?) -> TypeDescription {
+    if let parent = parent {
+      return .member(name: name, baseType: parent)
+    } else {
+      return .simple(name: name)
+    }
+  }
+
+  private var caseDescription: String {
+    switch self {
+    case .array:
+      return Self.arrayDescription
+    case .composition:
+      return Self.compositionDescription
+    case .dictionary:
+      return Self.dictionaryDescription
+    case .implicitlyUnwrapped:
+      return Self.implicitlyUnwrappedDescription
+    case .member:
+      return Self.memberDescription
+    case .optional:
+      return Self.optionalDescription
+    case .simple:
+      return Self.simpleDescription
+    case .tuple:
+      return Self.tupleDescription
+    case .unknown:
+      return Self.unknownDescription
+    }
+  }
+
+  private static let simpleDescription = "simple"
+  private static let memberDescription = "member"
+  private static let compositionDescription = "composition"
+  private static let optionalDescription = "optional"
+  private static let implicitlyUnwrappedDescription = "implicitlyUnwrapped"
+  private static let arrayDescription = "array"
+  private static let dictionaryDescription = "dictionary"
+  private static let tupleDescription = "tuple"
+  private static let unknownDescription = "unknown"
+}
+
+extension TypeSyntax {
```

Comment:
This extension works similarly to the now-deleted extension in `TypeSyntax+NameExtractor.swift`, but has more cases.